src/webui/api/torrentscontroller.cpp

```diff
@@ -1871,6 +1876,10 @@ void TorrentsController::renameFolderAction()
 {
     requireParams({u"hash"_s, u"oldPath"_s, u"newPath"_s});
 
+    const QString newFolderName = QFileInfo(params()[u"newPath"_s]).fileName();
```

Comment:
```suggestion
    const QString newFolderName = Path(params()[u"newPath"_s]).fileName();
```

Reply:
> Path has issues constructing paths with certain invalid filenames, such as "." and ".."

What do you mean exactly? Could you describe it in detail with some example path strings?

Reply:
> We can see that `Path.filename() `incorrectly returns the parent folder instead of the filename.

Ah, I get it. But there's nothing inherently "incorrect" with the path handling itself. `Path` constructor just simplifies the input path (particularly by resolving `.` and `..` whenever possible).